File: src/main/java/com/jozufozu/exnihiloomnia/common/lib/LibBlocks.java
package com.jozufozu.exnihiloomnia.common.lib;

import com.jozufozu.exnihiloomnia.ExNihilo;
import net.minecraft.util.ResourceLocation;

public class LibBlocks
{
    //Modifiers
    private static final String SUFFIX_WOOD = "_wood";
    private static final String SUFFIX_STONE = "_stone";
    private static final String SUFFIX_GLASS = "_glass";
    private static final String SUFFIX_CONCRETE = "_concrete";
    private static final String SUFFIX_TERRACOTTA = "_terracotta";
    private static final String STAINED = "_stained";
    
    //Blocks
    private static final String BARREL = "barrel";
    
    public static final ResourceLocation SIEVE = get("sieve");
    public static final ResourceLocation WOODEN_BARREL = get(BARREL, SUFFIX_WOOD);
    public static final ResourceLocation STONE_BARREL = get(BARREL, SUFFIX_STONE);
    public static final ResourceLocation GLASS_BARREL = get(BARREL, SUFFIX_GLASS);
    public static final ResourceLocation STAINED_GLASS_BARREL = get(BARREL, STAINED, SUFFIX_GLASS);
    public static final ResourceLocation TERRACOTTA_BARREL = get(BARREL, SUFFIX_TERRACOTTA);
    public static final ResourceLocation CONCRETE_BARREL = get(BARREL, SUFFIX_CONCRETE);
    
    public static final ResourceLocation CRUCIBLE = get("crucible");
    public static final ResourceLocation RAW_CRUCIBLE = get("raw_crucible");
    
    public static final ResourceLocation DUST = get("dust");
    public static final ResourceLocation NETHER_GRAVEL = get("nether_gravel");
    public static final ResourceLocation END_GRAVEL = get("end_gravel");
    
    public static final ResourceLocation KINDLING = get("kindling");
    
    public static final ResourceLocation INFESTED_LEAVES = get("infested_leaves");
    
    private static ResourceLocation get(String... name)
    {
        StringBuilder builder = new StringBuilder();
        for (String s : name)
        {
            builder.append(s);
        }
        
        return new ResourceLocation(ExNihilo.MODID, builder.toString());
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/lib/LibBlocks.kt
package com.jozufozu.exnihiloomnia.common.lib

import com.jozufozu.exnihiloomnia.ExNihilo
import net.minecraft.util.ResourceLocation

object LibBlocks {
    //Modifiers
    const val WOOD = "wood"
    const val STONE = "stone"
    const val GLASS = "glass"
    const val CONCRETE = "concrete"
    const val TERRACOTTA = "terracotta"
    const val STAINED = "stained"

    //Blocks
    const val BARREL = "barrel"

    val SIEVE = get("sieve")
    val WOODEN_BARREL = get(BARREL, WOOD)
    val STONE_BARREL = get(BARREL, STONE)
    val GLASS_BARREL = get(BARREL, GLASS)
    val STAINED_GLASS_BARREL = get(BARREL, STAINED, GLASS)
    val TERRACOTTA_BARREL = get(BARREL, TERRACOTTA)
    val CONCRETE_BARREL = get(BARREL, CONCRETE)

    val CRUCIBLE = get("crucible")
    val RAW_CRUCIBLE = get("raw_crucible")

    val DUST = get("dust")
    val NETHER_GRAVEL = get("nether_gravel")
    val END_GRAVEL = get("end_gravel")

    val INFESTED_LEAVES = get("infested_leaves")
    val WITCHWATER = get("witchwater")

    private fun get(vararg name: String): ResourceLocation {
        return ResourceLocation(ExNihilo.MODID, name.joinToString("_"))
    }
}
